pkg/errors/errors.go

```diff
@@ -0,0 +1,62 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+)
+
+type Error struct {
+	code    string
+	message string
+	err     error
+	params  []interface{}
+}
+
+func New(code, message string) *Error {
+	return &Error{
+		code:    code,
+		message: message,
+	}
+}
+
+// It returns true if the error is the same as the given error
+// it checks the error codes for comparison
+func Is(err1, err2 error) bool {
+	if err1 == nil || err2 == nil {
+		return false
+	}
+	e1, ok1 := err1.(*Error)
+	e2, ok2 := err2.(*Error)
+	return ok1 && ok2 && e1.code == e2.code
+}
```

Comment:
You shouldn't need to define this, you should use the standard go library `errors.Is` function https://pkg.go.dev/errors#Is

Reply:
right, that would be more idiomatic

Reply:
done